From 5830e89151e8304c6c842dd0f20a39a1b4275a56 Mon Sep 17 00:00:00 2001
From: Jason Paryani <github@jparyani.com>
Date: Fri, 4 Mar 2016 15:19:53 -0800
Subject: [PATCH 3/5] Increase max upload size to 100MB

---
 .user.ini | 5 +++++
 1 file changed, 5 insertions(+)
 create mode 100644 .user.ini

diff --git a/.user.ini b/.user.ini
new file mode 100644
index 00000000000..935fec5e7c7
--- /dev/null
+++ b/.user.ini
@@ -0,0 +1,5 @@
+max_execution_time=1800
+post_max_size=1000M
+upload_max_filesize=1000M
+max_input_time=1800
+max_file_uploads=300
-- 
2.39.2

