From 2ba2644755da531b0e94e1c9f71a9e4213516e68 Mon Sep 17 00:00:00 2001
From: Ian Denhardt <ian@zenhack.net>
Date: Sun, 20 Mar 2022 22:07:53 -0400
Subject: [PATCH 4/5] Set the URL bar using postMessage

---
 resources/src/startup/startup.js | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/resources/src/startup/startup.js b/resources/src/startup/startup.js
index e2452c4feaf..072db295afc 100644
--- a/resources/src/startup/startup.js
+++ b/resources/src/startup/startup.js
@@ -201,3 +201,8 @@ if ( !isCompatible( navigator.userAgent ) ) {
 		};
 	}() );
 }
+
+// SANDSTORM: update the url bar outside the iframe:
+window.parent.postMessage({
+  setPath: location.pathname + location.search,
+}, '*');
-- 
2.39.2

