From b455575ab78bf1b5fab13798c2b2cb0a6b99e1f2 Mon Sep 17 00:00:00 2001
From: Jason Paryani <github@jparyani.com>
Date: Wed, 13 Aug 2014 23:11:04 -0700
Subject: [PATCH 05/13] Add logic for users with same username

---
 includes/MediaWiki.php | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/includes/MediaWiki.php b/includes/MediaWiki.php
index 4ce07de849..23e9c0603d 100644
--- a/includes/MediaWiki.php
+++ b/includes/MediaWiki.php
@@ -191,7 +191,9 @@ class MediaWiki {
 		$user = $this->context->getUser();
 
 		$name = array_key_exists('HTTP_X_SANDSTORM_USERNAME', $_SERVER) ? $_SERVER[ 'HTTP_X_SANDSTORM_USERNAME'] : 'Public';
+		$original_name = $name;
 		$id = array_key_exists('HTTP_X_SANDSTORM_USER_ID', $_SERVER) ? $_SERVER[ 'HTTP_X_SANDSTORM_USER_ID'] : '';
+		$count = 2;
 
 		if ($user->isAnon()) {
 			$u = null;
@@ -204,8 +206,9 @@ class MediaWiki {
 				}
 
 				if (0 != $u->idForName() && $u->getEmail() !== $id && !empty($id)) {
-					$name = $name . "2";
+					$name = $original_name . $count;
 				}
+				$count++;
 			} while (0 != $u->idForName() && $u->getEmail() !== $id && !empty($id));
 
 			if ( 0 == $u->idForName() ) {
-- 
2.35.1

