From 30c3a754af2f8f947b7046fc223cbaa28443cb98 Mon Sep 17 00:00:00 2001
From: Jason Paryani <github@jparyani.com>
Date: Mon, 2 Feb 2015 14:45:31 -0800
Subject: [PATCH 6/9] urldecode Sandstorm usernames

---
 includes/MediaWiki.php | 1 +
 1 file changed, 1 insertion(+)

diff --git a/includes/MediaWiki.php b/includes/MediaWiki.php
index 36cfde06d1..eee26457e8 100644
--- a/includes/MediaWiki.php
+++ b/includes/MediaWiki.php
@@ -227,6 +227,7 @@ class MediaWiki {
 		$user = $this->context->getUser();
 
 		$name = array_key_exists('HTTP_X_SANDSTORM_USERNAME', $_SERVER) ? $_SERVER[ 'HTTP_X_SANDSTORM_USERNAME'] : 'Public';
+		$name = urldecode($name);
 		$original_name = $name;
 		$id = array_key_exists('HTTP_X_SANDSTORM_USER_ID', $_SERVER) ? $_SERVER[ 'HTTP_X_SANDSTORM_USER_ID'] : '';
 		$count = 2;
-- 
2.35.1

