From cb85591175018c21ab2d96959c3d668fd30c3553 Mon Sep 17 00:00:00 2001
From: Jason Paryani <github@jparyani.com>
Date: Fri, 21 Aug 2015 16:34:51 -0700
Subject: [PATCH 7/9] Change Sandstorm auth to appropriately upgrade/downgrade
 permissions

---
 includes/MediaWiki.php | 19 +++++++++++++++++++
 1 file changed, 19 insertions(+)

diff --git a/includes/MediaWiki.php b/includes/MediaWiki.php
index eee26457e8..4f7151994a 100644
--- a/includes/MediaWiki.php
+++ b/includes/MediaWiki.php
@@ -270,6 +270,25 @@ class MediaWiki {
 					$u->saveSettings();
 				}
 				$isNew = true;
+			} else {
+				// Check that current user's groups are correct
+				$permissions = array_key_exists('HTTP_X_SANDSTORM_PERMISSIONS', $_SERVER) ? $_SERVER[ 'HTTP_X_SANDSTORM_PERMISSIONS'] : '';
+				$groups = $u->getGroups();
+				if (in_array("sysop", $groups)) {
+					if (strpos($permissions, 'admin') === false) {
+						wfLogWarning('removing user as admin: ' . $name);
+						$u->removeGroup( 'sysop' );
+						$u->removeGroup( 'bureaucrat' );
+						$u->saveSettings();
+					}
+				} else {
+					if (strpos($permissions, 'admin') !== false) {
+						wfLogWarning('setting user as admin: ' . $name);
+						$u->addGroup( 'sysop' );
+						$u->addGroup( 'bureaucrat' );
+						$u->saveSettings();
+					}
+				}
 			}
 			// wfSetupSession();
 			// Not sure why, but I manually have to set session cookie
-- 
2.35.1

